apis/kueue/v1beta1/workload_types.go

```diff
@@ -514,7 +514,7 @@ const (
 
 	// WorkloadEvictedByDeactivation indicates that the workload was evicted
 	// because spec.active is set to false.
-	WorkloadEvictedByDeactivation = "InactiveWorkload"
+	WorkloadEvictedByDeactivation = "Deactivated"
```

Comment:
This is great, but might be problematic on upgrade, because we have in our code places which check, for example:
- [here](https://github.com/kubernetes-sigs/kueue/blob/b0665c7d41b998d490e1f5f8a15dba278836868e/pkg/workload/workload.go#L771) 
- [here](https://github.com/kubernetes-sigs/kueue/blob/b0665c7d41b998d490e1f5f8a15dba278836868e/pkg/controller/core/workload_controller.go#L163)

possibly more - we will see them all if you rename the const too, to `WorkloadDeactivated`.

IIUC changing the name will impact the workloads, and we need to make sure the upgrade is smooth, so I would suggest:
- introduce the new reason, and set it 
- instead of removing the old reason, deprecate it, but support it if the old reason is set 

We will remove the old reason in one or two releases. WDYT @PBundyra @tenzen-y ?

Reply:
We have another problem https://github.com/kubernetes-sigs/kueue/blob/8f9b6dc58761833e8d194a7e480a262dd22bb023/pkg/controller/core/workload_controller_test.go#L1121

In that case it will be EvictedDueToDeactivatedDueToRequeuingLimitExceeded

Reply:
Oh, not correct thread